### ISSUE TITLE
Restrictive pragma makes it difficult to use this repository

### ISSUE THREAD

**Issue by paulrberg** [mds1/multicall#53]
Unless I am misunderstanding how I should this, I think that the current pragma is unnecessarily restrictive by being locked to `0.8.12`. I am getting this compiler error in my project, which uses v0.8.17:

```solidity
Compiler run failed
error[5333]: ParserError: Source file requires different compiler version (current compiler is 0.8.17+commit.8df45f5f.Darwin.appleclang) - note that nightly builds are considered to be strictly less than the released version
 --> lib/multicall/src/Multicall3.sol:2:1:
  |
2 | pragma solidity 0.8.12;
  | ^^^
```

Is there any specific reason for locking in the pragma to v0.8.12 and not writing it like this?

```solidity
pragma solidity >=0.8.12 <0.9.0;
```

It also happens that this is a version that contains [an important bug related to `abi.encodeCall`](https://blog.soliditylang.org/2022/03/16/solidity-0.8.13-release-announcement/), which is highly relevant in the context of Multicall.

**Comment by mds1** [comment]
What are you trying to do by importing the multicall contract? The contract's pragma is locked to 0.8.12 to ensure all networks have the same resulting bytecode when it's deployed. I'm guessing you just want the interface, in which case perhaps we should add a `src/interfaces/IMulticall.sol` file with a looser pragma?

**Comment by paulrberg** [comment]
I came here from the [Best Practices](https://book.getfoundry.sh/tutorials/best-practices#fork-tests) section of the Foundry Book, where it recommended to use Multicall (specifically, this repository) to replace multiple RPC calls with a single call.

Re pragma lock - oh, I see, thanks for explaining. That makes sense to me.

You're right, I simply want the interface. I would prefer to be able to source it from this repository rather than maintain a copy-pasted version in my own repository.

I have opened a PR for this: https://github.com/mds1/multicall/pull/55
